_posts/2022-06-15-ejb-over-https.adoc

```diff
@@ -0,0 +1,340 @@
+----
+
+Finally, we create a key manager, which will be used to access the key store during the TLS handshake:
+
+[source]
+----
+/subsystem=elytron/key-manager=sslKM:add(key-store=sslKeyStore,credential-reference={clear-text=serverKeySecret})
+----
+
+== Importing the Certificates
+
+Since we are already connected to the server, we will import the client certificate first. We configure a new trust store for holding certificates:
+
+[source]
+----
+/subsystem=elytron/key-store=sslTrustStore:add(path=sslServer.truststore,relative-to=jboss.server.config.dir,credential-reference={clear-text=serverTrustSecret})
+----
+
+We import the client certificate, specifying it with its alias, and then save it to the trust store file. Since this is
+a self-signed certificate, we will not validate its authenticity.
+
+[source]
+----
+/subsystem=elytron/key-store=sslTrustStore:import-certificate(alias=client,path=/PATH/TO/sslClient.cer,credential-reference={clear-text=serverTrustSecret},trust-cacerts=true,validate=false)
+
+/subsystem=elytron/key-store=sslTrustStore:store()
+----
+
+We also configure a trust manager, which will access the certificate during the TLS handshake:
+
+[source]
+----
+/subsystem=elytron/trust-manager=sslTM:add(key-store=sslTrustStore)
+----
+
+With the server now setup, we return to the client. Open a new terminal and navigate to the root folder of this
+example. We use the *keytool* once again, identify the server's certificate by its alias and importing it into a client
+trust store:
+
+[source,shell]
+----
+$ keytool -importcert -keystore /PATH/TO/sslClient.truststore -storepass clientTrustSecret -alias localhost -trustcacerts -file /WILDFLY_HOME/standalone/configuration/sslServer.cer -noprompt
+----
+
+At this point, the client and server both have their corresponding key pairs, and trust the certificates of the other.
+The TLS configuration is complete.
+
+== Configuring the Credential Store
+
+We will now setup the client login using the SASL authentication mechanism. First, we use the *elytron-tool* generate a
+credential store to securely save the login details:
+
+[source,shell]
+----
+$ WILDFLY_HOME/bin/elytron-tool.sh credential-store --create --location "/PATH/TO/sslCredStore.cs" --password clientStorePassword
+----
+
+Then, we add the `example_user` identity to this credential store:
+
+[source,shell]
+----
+$ WILDFLY_HOME/bin/elytron-tool.sh credential-store --location "/PATH/TO/sslCredStore.cs" --password clientStorePassword --add example_user --secret examplePwd1!
+----
+
+== Configuring the WildFly Client
+
+Now that we have our key store, trust store, and credential store configured, we need to tell the WildFly client how to
+access them. From the root of this example, navigate to the directory at `src/main/resources` and open the
+`wildfly-config.xml` configuration file in a text editor. Upon inspecting the file, we note that the credential store
+is imported into the client to access credentials:
```

Comment:
Just a small terminology thing, instead of "importing", we should reword this to something like "note the credential store configuration. This will be used to access the client's credential when establishing a connection to the server" or something along those lines.